### PR TITLE
Fibonacci: Converting to Text Template

### PR DESCRIPTION
**Pull request by mintsoft**
###### Description of changes

Ref: https://github.com/duckduckgo/zeroclickinfo-goodies/issues/2767

I've also extended the test coverage around question formats that were previously untested

###### People to notify (@mention interested parties)

---

Instant Answer Page: https://duck.co/ia/view/fibonacci

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @koosha--

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Goodie/Fibonacci.pm [hunk 1]
```diff
@@ -16,23 +16,26 @@ my @fib = (0, 1);
 handle remainder => sub {
     s/^\s+//;
     s/\s+$//;
-    return unless /^(?:what(?:'s| is) the )?(?<which>\d+)(?:th|rd|st)?(?: number)?(?: in the (?:series|sequence))?\??$/ && $1 <= 1470;
+    return unless /^(?:what(?:'s| is) the )?(?<which>\d+)(?:th|rd|st)?(?: number)?(?: in the (?:series|sequence))?\??$/ && $1 <= 1470; #' (hack for broken syntax highlighting)
```
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/Fibonacci.pm]: What's this?

Comment (mintsoft) [inline lib/DDG/Goodie/Fibonacci.pm]: This is my high quality solution for the `'` in the `qr//` confusing certain syntax highlighting systems :+1:

Comment (GuiltyDolphin) [inline lib/DDG/Goodie/Fibonacci.pm]: I think we should remove it - you can add it as a temp fix when working on it but it's not a global issue (I see no syntax highlighting issues).

File: t/Fibonacci.t [hunk 2]
```diff
@@ -8,24 +8,27 @@ use DDG::Test::Goodie;
...
-            input     => ['33rd'],
-            operation => 'Fibonacci number',
-            result    => 3524578
-        }
-    ),
+    'fib 7' => build_test('The 7th fibonacci number is 13 (assuming f(0) = 0).', 13, '7th'),
+    'fibonacci 33' => build_test('The 33rd fibonacci number is 3524578 (assuming f(0) = 0).', 3524578, '33rd'),
+    q`what's the 6th fibonacci number?` => build_test('The 6th fibonacci number is 8 (assuming f(0) = 0).', 8, '6th'),
+    '10th number in the fibonacci sequence?' => build_test('The 10th fibonacci number is 55 (assuming f(0) = 0).', 55, '10th'),
+    'what is the 18th number in the fibonacci series' => build_test('The 18th fibonacci number is 2584 (assuming f(0) = 0).', 2584, '18th'),
```
Comment (GuiltyDolphin) [inline t/Fibonacci.t]: We don't need to pass in the full text, all the info is contained within the other two values.

**Comment by GuiltyDolphin** [comment]
@mintsoft Thanks! :dolphin:

**Pull request merged**